### PR TITLE
🐛 Source Facebook-Marketing: updated facebook_business==17.0.0

### PR DESCRIPTION
**Pull request by ysusuk**
[Issue](https://github.com/airbytehq/airbyte/issues/26745)

### CODE REVIEW

File: airbyte-ci/connectors/metadata_service/orchestrator/tests/fixtures/cloud_registry.json [hunk 1]
```diff
@@ -8743,7 +8743,7 @@
       "sourceDefinitionId": "e7778cfc-e97c-4458-9ecb-b4f2bba8946c",
       "name": "Facebook Marketing",
       "dockerRepository": "airbyte/source-facebook-marketing",
-      "dockerImageTag": "0.2.86",
+      "dockerImageTag": "0.4.3",
```
Comment (marcosmarxm) [inline airbyte-ci/connectors/metadata_service/orchestrator/tests/fixtures/cloud_registry.json:8746]: These files are edited automatically please rollback the changes here you only need to change the version in the connector folder.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-config-oss/init-oss/src/test/resources/connector_catalog.json [hunk 2]
```diff
@@ -6604,7 +6604,7 @@
       "sourceDefinitionId": "e7778cfc-e97c-4458-9ecb-b4f2bba8946c",
       "name": "Facebook Marketing",
       "dockerRepository": "airbyte/source-facebook-marketing",
-      "dockerImageTag": "0.2.60",
+      "dockerImageTag": "0.2.86",
```
Comment (marcosmarxm) [inline airbyte-config-oss/init-oss/src/test/resources/connector_catalog.json:6607]: Revert here.

File: airbyte-ci/connectors/metadata_service/orchestrator/tests/fixtures/oss_registry.json [hunk 3]
```diff
@@ -13179,7 +13179,7 @@
       "sourceDefinitionId": "e7778cfc-e97c-4458-9ecb-b4f2bba8946c",
       "name": "Facebook Marketing",
       "dockerRepository": "airbyte/source-facebook-marketing",
-      "dockerImageTag": "0.2.86",
+      "dockerImageTag": "0.4.3",
```
Comment (marcosmarxm) [inline airbyte-ci/connectors/metadata_service/orchestrator/tests/fixtures/oss_registry.json:13182]: and here.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Review by marcosmarxm — APPROVED** [review]
Thanks I'm going to start running the integration tests.

**Comment by ysusuk** [comment]
GM, @marcosmarxm!
Is there anything else i can do?

**Comment by marcosmarxm** [comment]
Running tests in https://github.com/airbytehq/airbyte/pull/27412

**Review by davydov-d — APPROVED** [review]

**Comment by marcosmarxm** [comment]
Tests for Facebook Marketing are failing in main branch. We must need wait until they get fixed to release this contribution.

**Comment by danilo-nzyte** [comment]
@ysusuk @marcosmarxm I left a comment on the #27412 about the error. It looks like it's due to a missing `token` value - how do we solve this?

P.S. Thank you for the contribution, I really need this fix.

**Comment by marcosmarxm** [comment]
@ysusuk sorry the delay. The integration tests were fixed but looks version 17 of the Facebook SDK introduced a breaking change:
```
3007.6 ***"type": "LOG", "log": ***"level": "WARN", "message": "Unable to parse connector's output /usr/local/lib/python3.9/site-packages/facebook_business/utils/api_utils.py:30: UserWarning:  does not allow field has_advertiser_opted_in_odax\n, error: 1 validation error for AirbyteMessage\n__root__\n  Expecting value: line 1 column 1 (char 0) (type=value_error.jsondecode; msg=Expecting value; doc=/usr/local/lib/python3.9/site-packages/facebook_business/utils/api_utils.py:30: UserWarning:  does not allow field has_advertiser_opted_in_odax\n; pos=0; lineno=1; colno=1)"***
#11 3007.6 ***"type": "LOG", "log": ***"level": "WARN", "message": "Unable to parse connector's output   warnings.warn(message)\n, error: 1 validation error for AirbyteMessage\n__root__\n  Expecting value: line 1 column 3 (char 2) (type=value_error.jsondecode; msg=Expecting value; doc=  warnings.warn(message)\n; pos=2; lineno=1; colno=3)"***
#11 3007.6 ***"type": "LOG", "log": ***"level": "ERROR", "message": "Docker container failed, code 1, error:\n***\"type\": \"TRACE\", \"trace\": ***\"type\": \"ERROR\", \"emitted_at\": 1687284928671.694, \"error\": ***\"message\": \"Something went wrong in the connector. See the logs for more details.\", \"internal_message\": \"\\n\\n  Message: Call was not successful\\n  Method:  GET\\n  Path:    [https://graph.facebook.com/v17.0/act_***/\\n](https://graph.facebook.com/v17.0/act_***///n)  Params:  ***'fields': 'account_id,account_status,age,amount_spent,balance,business,business_city,business_country_code,business_name,business_state,business_street,business_street2,business_zip,can_create_brand_lift_study,capabilities,created_time,currency,disable_reason,end_advertiser,end_advertiser_name,extended_credit_invoice_group,failed_delivery_checks,fb_entity,funding_source,has_advertiser_opted_in_odax,has_migrated_permissions,id,io_number,is_attribution_spec_system_default,is_direct_deals_enabled,is_in_3ds_authorization_enabled_market,is_notifications_enabled,is_personal,is_tax_id_required,line_numbers,media_agency,min_campaign_group_spend_cap,min_daily_budget,name,offsite_pixels_tos_accepted,owner,partner,rf_spec,spend_cap,tax_id,tax_id_status,tax_id_type,timezone_id,timezone_name,timezone_offset_hours_utc,tos_accepted,user_tasks,user_tos_accepted'***\\n\\n  Status:  400\\n  Response:\\n    ***\\n      \\\"error\\\": ***\\n        \\\"message\\\": \\\"(#12) Since we have ODAX enforcement from v17, this field is useless and should always be assumed true is deprecated for versions v17.0 and higher\\\",\\n        \\\"type\\\": \\\"OAuthException\\\",\\n        \\\"code\\\": 12,\\n        \\\"fbtrace_id\\\": \\\"A608Sfa9B4-w3XZRbS-EErZ\\\"\\n      ***\\n    ***\\n\", \"stack_trace\": \"Traceback (most recent call last):\\n  File \\\"/airbyte/integration_code/main.py\\\", line 13, in <module>\\n    launch(source, sys.argv[1:])\\n  File \\\"/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py\\\", line 171, in launch\\n    for message in source_entrypoint.run(parsed_args):\\n  File \\\"/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py\\\", line 101, in run\\n    yield from map(AirbyteEntrypoint.airbyte_message_to_string, self.read(source_spec, config, config_catalog, state))\\n  File \\\"/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py\\\", line 141, in read\\n    yield from self.source.read(self.logger, config, catalog, state)\\n  File \\\"/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py\\\", line 141, in read\\n    raise e\\n  File \\\"/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py\\\", line 121, in read\\n    yield from self._read_stream(\\n  File \\\"/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py\\\", line 196, in _read_stream\\n    for record in record_iterator:\\n  File \\\"/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py\\\", line 333, in _read_full_refresh\\n    for record_data_or_message in record_data_or_messages:\\n  File \\\"/airbyte/integration_code/source_facebook_marketing/streams/base_streams.py\\\", line 113, in read_records\\n    for record in loaded_records_iter:\\n  File \\\"/airbyte/integration_code/source_facebook_marketing/streams/base_streams.py\\\", line 109, in <genexpr>\\n    loaded_records_iter = (record.api_get(fields=self.fields, pending=self.use_batch) for record in records_iter)\\n  File \\\"/usr/local/lib/python3.9/site-packages/facebook_business/adobjects/adaccount.py\\\", line 260, in api_get\\n    return request.execute()\\n  File \\\"/usr/local/lib/python3.9/site-packages/facebook_business/api.py\\\", line 682, in execute\\n    response = self._api.call(\\n  File \\\"/usr/local/lib/python3.9/site-packages/backoff/_sync.py\\\", line 105, in retry\\n    ret = target(*args, **kwargs)\\n  File \\\"/airbyte/integration_code/source_facebook_marketing/api.py\\\", line 154, in call\\n    response = super().call(method, path, params, headers, files, url_override, api_version)\\n  File \\\"/usr/local/lib/python3.9/site-packages/facebook_business/api.py\\\", line 350, in call\\n    raise fb_response.error()\\nfacebook_business.exceptions.FacebookRequestError: \\n\\n  Message: Call was not successful\\n  Method:  GET\\n  Path:    [https://graph.facebook.com/v17.0/act_***/\\n](https://graph.facebook.com/v17.0/act_***///n)  Params:  ***'fields':
```

During the day I'll investigate to see what need to be changed to fix the problem.

**Comment by marcosmarxm** [comment]
@ysusuk the change will be more complex than expected. I asked @davydov-d to handle all the modifications needed to get the connector to work with the new SDK 17.

**Comment by ysusuk** [comment]
Got it, thank you for the update @marcosmarxm!

**Comment by marcosmarxm** [comment]
@ysusuk I'm closing your contribution in favor of https://github.com/airbytehq/airbyte/pull/27563 which is handling the compatibility change. Thanks for the contribution 🎖️

**Pull request closed without merging**